### PR TITLE
Refactoring workflows to have reusable jobs

### LINKED ISSUE #2602: Refactor GitHub Actions workflows to use Reusable Workflows
Our current CI and Release workflows duplicate a bunch of things.  It would be good if we could have the deploy reuse things from CI, or form a base file.  This is now supported in GitHub Actions, see https://docs.github.com/en/actions/learn-github-actions/reusing-workflows

### PR DESCRIPTION
**Pull request by dhillonks**
## Issue This PR Addresses
Closes #2602 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [X] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Created the following reusable workflows after refactoring `node-js-ci.yml` and `release.yml`:
- `.github/workflows/e2e-ci.yml`
- `.github/workflows/eslint-ci.yml`
- `.github/workflows/prettier-ci.yml`
- `.github/workflows/unit-tests-ci.yml`

Note: The repo URL for referencing the reusable workflow in `uses` is set to my fork as that is where the files currently reside. This will be changed to `Seneca-CDOT/telescope` once the pr is ready for a merge.

Running as expected after testing the runs on my fork:
image

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: N/A
- [ ] **Documentation**: N/A

### CODE REVIEW

File: .github/workflows/e2e-ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Reusable e2e Workflow
```
Comment (humphd) [inline .github/workflows/e2e-ci.yml:1]: Let's rename this file `e2e-tests.yml` to mimic the unit tests file naming.

**Review by humphd — CHANGES_REQUESTED** [review]
This is awesome!  Can you get rid of the word `Reusable` in all the names?  It's redundant. One naming issue to fix, and can you correct all those URLs to point to this repo?  After that, I think it's ready.

File: .github/workflows/e2e-ci.yml (hunk 1 above)
Comment (dhillonks) [reply .github/workflows/e2e-ci.yml]: Done!

**Comment by dhillonks** [comment]
Done!

To prepare for the merge, I've updated the URL to this repo and also the `ref` to `master` (Please note that the newly triggered run in this pr will fail as the workflow files don't exist yet in the `ref`)

**Review by humphd — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Comment by humphd** [comment]
I'll merge this and we'll see if it breaks on `master`...

**Comment by humphd** [comment]
Here's the run now: https://github.com/Seneca-CDOT/telescope/actions/runs/1585498260

**Comment by humphd** [comment]
Green ✅

**Pull request merged**